### ISSUE TITLE
Allow specific MaxLineLength for individual cops

### ISSUE THREAD

**Issue by CyborgMaster** [rubocop-hq/rubocop#6077]
The option for individual `MaxLineLength`s for specific cops used to be an option and was removed here:  https://github.com/rubocop-hq/rubocop/commit/9e76387952725f1a7fa54989024114549c0b7d80#diff-262d848b3b481343a51bf597d6f4d574L44.  I agree that removing it from default.yml is critical because that causes unexpected behavior, but removing the option entirely reduces RuboCop's flexibility.

In our company we set `LineLength`'s `Max` setting to 89.  The company's official guide states 80 characters, but passing RuboCop is a requirement for merging code, so we allow up to 89 to provide a little leeway and reduce resistance to following the style requirements. 

However, because the official guide for the company is 80, we don't want other cops requiring changes that would force the line length over 80.  Previously setting the `MaxLineLength` option for the `IfUnlessModifier` cop to 80 would prevent that cop flagging an offense on code that would be over 80 characters when converted to a single line modifier.  Now I have no way to handle this situation.  

Therefore I propose re-adding the following to `StatmentModifier`:

``` ruby
def max_line_length
  cop_config['MaxLineLength'] ||
    config.for_cop('Metrics/LineLength')['Max']
end
``` 

While keeping those options out of default.yml, so the default behaivor is to follow `Max` from `LineLength`.

If this is agreeable, I can provide a quick PR.

--------

## Expected behavior

No offense

## Actual behavior

```
test/test_helper.rb:363:7: C: Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||. (https://github.com/rubocop-hq/ruby-style-guide#if-as-a-modifier)
      if Time.now - wait_start > 2
      ^^
```

## Steps to reproduce the problem

Have the following config:
``` yaml
LineLength: { Max: 89 }
Style/IfUnlessModifier: { MaxLineLength: 80 }
```
and the following code:
``` ruby
      if Time.now - wait_start > 2
        raise "Expected email to have been sent to #{address}"
      end
```

## RuboCop version

```
➜ rubocop -v 
0.57.2
```

**Comment by CyborgMaster** [comment]
Updated to include the output of `rubocop -V` instead of `rubocop -v`

**Comment by MikeMcQuaid** [comment]
Homebrew would still find this useful so just commenting so stale bot knows someone cares (although this could be combined with https://github.com/rubocop-hq/rubocop/issues/3534)

**Comment by CyborgMaster** [comment]
I would still love this as well

**Comment by CyborgMaster** [comment]
I would still like to see this implimented

**Comment by biinari** [comment]
A workaround is possible, as `IgnoredPatterns` defined in `Layout/LineLength` cop do not affect the other cops. In this configuration, `Layout/LineLength` will allow lines up to 89 characters long, but `Style/IfUnlessModifier` and other cops will treat the limit as 80 characters.

```yaml
Layout/LineLength:
  Max: 80
  IgnoredPatterns:
    - '^.{,89}$'
```

**Comment by mvz** [comment]
Since the default `Max:` for `Layout/LineLength` has been increased to 120, this would be very useful to avoid extra long lines with If/Unless modifiers.

**Comment by CodingAnarchy** [comment]
This feature would also be useful for cases where we want a larger overall line limit (we use 160 characters ourselves), but would not like single-line conditionals of that length.
